### ISSUE TITLE
Loss of IP Alarm on mux inputs

### ISSUE THREAD

**Issue by nickpiggottwork** [smallscaledab/small-scale-dab#10]
If the mux loses an IP stream from an input source, for longer than a configurable amount of time, it should distribute an alert to configured destination(s) for that input source.

**Comment by mpbraendli** [comment]
That can be done though the Statistics/Management server combined with the munin plugin.

It's not a feature I want to see inside ODR-DabMux, because it can be quite complex, and if this fails it could crash the whole multiplexer. But the information is available, and a small script could take care of sending warnings if the munin setup is not used.

I should document this better :-/

**Comment by nickpiggottwork** [comment]
I agree it's possible through the stats plug in. I'm wondering if there's something that could sit on the ZMQ inputs and if the buffers drop to 0 for a certain amount of time, it logs/triggers. That would be less invasive than complete stats/management and hopefully less destabilising?

Even if it writes to a log file that the buffer went to 0 at time HH;MM;SS, then something like Zenoss could be used to create emails alerts and so on.

**Comment by mpbraendli** [comment]
These options come to mind:
- A small script that regularly asks the buffer sizes through the stats server, and sends warnings accordingly.
- Log to syslog, and analyse the system log with an appropriate tool. Parse for the "inputZMQ <name> input pre-buffering complete" and "inputZMQ <name> input empty, re-enabling pre-buffering" messages, that get output when the ZMQ buffer state changes.

Both can make more sense than setting up munin.
